File: mang5o/220214/bj-9095.py
# https://www.acmicpc.net/problem/9095
def get_output(now_figure):
    if now_figure < 0:
        return 0
    elif now_figure <= 1:
        return 1
    else:
        return get_output(now_figure - 1) + get_output(now_figure - 2) + get_output(now_figure - 3)


outputs = []
for i in range(int(input())):
    outputs.append(get_output(int(input())))
for i in outputs:
    print(i)

File: adultlee/220214/bj9095.cpp
// https://www.acmicpc.net/problem/9095
// 1 ,2 ,3 숫자 더하기

#include <iostream>

using namespace std;


int dp[101];
int main(){

    int case;
    cin << case;

    dp[1]=1;
    dp[2]=2;
    dp[3]=4;
    int T;
    for(int i=0; i<case; i++)
    {   cin << T;
        for(int j=4; j<11; j++)
        {
            dp[j] = dp[j-3] + dp[j-2] + dp[j-1];
        }
        cout << dp[T];
    }
    
}